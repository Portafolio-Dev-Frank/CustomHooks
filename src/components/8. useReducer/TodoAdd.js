import React from 'react'
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAdd }) => {

    const [{ description }, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleOnSubmit = (e) => {
        e.preventDefault();
        if (description.trim().length <= 1) {
            return
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }
        handleAdd(newTodo);
        reset();
    }

    return (
        <>
            <h4>Agregar Todo</h4>
            <hr></hr>
            <form onSubmit={handleOnSubmit}>
                <div class="d-grid gap-2">
                    <input
                        type="text"
                        name="description"
                        placeholder="aprender"
                        autoComplete="off"
                        value={description}
                        onChange={handleInputChange}
                    >
                    </input>
                    <button type="submit" className="btn btn-outline-info mt-1">
                        agregar
                        </button>
                </div>
            </form>
        </>
    )
}
