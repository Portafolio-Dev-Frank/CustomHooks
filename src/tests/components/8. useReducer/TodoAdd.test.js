import { shallow } from "enzyme"
import { TodoAdd } from "../../../components/8. useReducer/TodoAdd"


describe('pruebas en <TodoAdd></TodoAdd>', () => {

    const handleAdd = jest.fn();
    const wrapper = shallow(<TodoAdd
        handleAdd={handleAdd}
    ></TodoAdd>)

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('no debe de llamar addTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault() {}})
        expect(handleAdd).toHaveBeenCalledTimes(0);
    })

    test('debe de llamar la funcion handleAdd', () => {
        const input = wrapper.find('input');
        const value = 'valor de formulario';
        input.simulate('change', { target: { value, name: 'description' } });
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault() {}})
        expect(handleAdd).toHaveBeenCalled();
        expect(handleAdd).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAdd).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc:value,
            done: false
        });
        expect(wrapper.find('input').prop('value')).toBe('')
    })
    
    
})
