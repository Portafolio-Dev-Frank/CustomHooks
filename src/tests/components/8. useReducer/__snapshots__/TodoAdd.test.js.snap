// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoAdd></TodoAdd> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      class="d-grid gap-2"
    >
      <input
        autoComplete="off"
        name="description"
        onChange={[Function]}
        placeholder="aprender"
        type="text"
        value=""
      />
      <button
        className="btn btn-outline-info mt-1"
        type="submit"
      >
        agregar
      </button>
    </div>
  </form>
</Fragment>
`;
